DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/README.md

```diff
@@ -0,0 +1,392 @@
+the `VERBOSE=1` argument:
+``make VERBOSE=1``
+For more comprehensive troubleshooting, use the Diagnostics Utility for
+Intel® oneAPI Toolkits, which provides system checks to find missing
+dependencies and permissions errors.
+[Learn more](https://software.intel.com/content/www/us/en/develop/documentation/diagnostic-utility-user-guide/top.html).
+
+ ### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this tutorial in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [Intel® oneAPI DPC++ FPGA Workflows on Third-Party IDEs]([https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-fpga-workflow-on-ide](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-oneapi-dpcpp-fpga-workflow-on-ide.html))
+
+## Examining the Reports
+
+Locate `report.html` in the corresponding `<source_file>_report.prj/reports/` directory. Open the report in any of the following web browsers:  Chrome*, Firefox*, Edge*, or Internet Explorer*.
+
+Open the **Views** menu and select **System Viewer**.
+
+In the left-hand pane, select **FunctorRegisterMapIP** or **LambdaRegisterMapIP** under the System hierarchy for the kernels with a register-mapped invocation interface.
+
+In the main **System Viewer** pane, the kernel invocation interfaces and kernel argument interfaces are shown. They show that the `start`, `busy` and `done` kernel invocation interfaces are implemented in register map interfaces, the `arg_input` and `arg_output` kernel arguments are implemented in register map interfaces. The `arg_n` kernel argument is implemented in a streaming interface in the **FunctorRegisterMapIP**, and in a register map interface in the **LambdaRegisterMapIP**.
+
+Similarly, in the left-hand pane, select **FunctorStreamingIP** or **LambdaStreamingIP** under the System hierarchy for the kernels with a streaming invocation interface.
+
+In the main **System Viewer** pane, the kernel invocation interface and kernel argument interfaces are shown. They show that the `start`, `done`, `ready_in` and `ready_out` kernel invocation interfaces are implemented in streaming interfaces, and the `arg_input`, `arg_output` and `arg_n` kernel arguments are all implemented in streaming interfaces.
+
+## Running the Sample
+
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU):
+     ```
+     ./register_map_functor_model.fpga_emu        (Linux)
+     ./streaming_functor_model.fpga_emu           (Linux)
+     ./register_map_lambda_model.fpga_emu         (Linux)
+     ./streaming_lambda_model.fpga_emu            (Linux)
+     register_map_functor_model.fpga_emu.exe      (Windows)
+     streaming_functor_model.fpga_emu.exe         (Windows)
+     register_map_lambda_model.fpga_emu.exe       (Windows)
+     streaming_lambda_model.fpga_emu.exe          (Windows)
+     ```
+2. Run the sample on the FPGA simulator:
```

Comment:
With the new selectors for 2023.1, we need to set an environment variable to be able to run the simulator, so there is the kind of thing we expect:

Linux:
   ```
   CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1 ./register_map_functor_model.fpga_sim
   ``` 

Windows:
   ```
   set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1
   register_map_functor_model.fpga_sim.exe
   set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=
   ```